build.sh

```diff
@@ -235,37 +247,52 @@ crontab -l
 EOF
 
 
-ssh $osname  "$VM_SHUTDOWN_CMD"
+if [ -e "hooks/sysUpdate.sh" ]; then
+  echo "hooks/sysUpdate.sh"
+  cat "hooks/sysUpdate.sh"
+  ssh $osname sh<"hooks/sysUpdate.sh"
 
-sleep 30
+  # We need to reboot after system updates
+  restart_and_wait
+fi
 
-###############################################################
 
-if $vmsh isRunning $osname; then
-  if ! $vmsh shutdownVM $osname; then
-    echo "shutdown error"
-  fi
+if [ -e "hooks/postBuild.sh" ]; then
+  echo "hooks/postBuild.sh"
+  cat "hooks/postBuild.sh"
+  ssh $osname sh<"hooks/postBuild.sh"
 fi
 
-while $vmsh isRunning $osname; do
-  sleep 5
-done
 
+# Install any requested packages
+if [ "$VM_PRE_INSTALL_PKGS" ]; then
+  echo "$VM_INSTALL_CMD $VM_PRE_INSTALL_PKGS"
+  ssh $osname sh <<<"$VM_INSTALL_CMD $VM_PRE_INSTALL_PKGS"
+fi
 
-##############################################################
+# Done!
+shutdown_and_wait
 
 
+##############################################################
 
+if [ "$VM_ISO_LINK" ]; then
+  echo "Clean up ISO for more space"
+  sudo rm -f ${osname}.iso
+fi
 
-ova="$osname-$VM_RELEASE.qcow2"
+echo "contents of home directory:"
+ls -lah
 
+echo "free space:"
+df -h
 
-echo "Exporting $ova"
+ova="$osname-$VM_RELEASE.qcow2.xz"
```

Comment:
don't change the ova value, keep it as before.
The exportOVA function would append ".xz" suffix automatically.

Reply:
oops, that's from when i was skipping the export command from vbox.sh